### DISCUSSION TITLE
BigQuery & Labels - Need help :)

### DISCUSSION THREAD

**Discussion by LasseLegarth** [apache/superset#29452] [Q&A / Help]
Hi,

Im bigquery it's possible to set a label for each query, I want to use this functionallity to "tag" each query to the superset user who have initiated the query. When doing it in plain SQL it can be done with the following SQL:
`SET @@query_label = "label_key:label_value";SELECT * FROM mytable`

The above works fine in the SQL lab, where I use jinja to set the label_value with {{ current_username() }}

The problem is there when I try to create a dataset from the above SQL, then I get an fatal error - since a virtual dataset can only have one statement. Yet I just want to have the SET to set the label and then the SELECT as the real statement.

I also tried to edit the database connection, there I can under Additional Parameters add labels=mylabel:mytest - which also works and sets the label. But I can't use jinja to set the username dynamically from there :(

Any ideas on how I can set a label on the queries I make with the username or something to identify the user? I'm really stuck here so any help or ideas will be more than welcome! Thanks :)

**Comment by migueldias71** [comment]
hello @LasseLegarth I have the same exact problem. Have you found a solution yet?
Thanks
